### PR TITLE
Replace StringBuilder#append(String) with StringBuilder#append(char)

### PR DESCRIPTION
**Pull request by smola**
# What Does This Do
Replace `StringBuilder#append(String)` with `StringBuilder#append(char)` when a single character is used.

# Motivation
Detected with [java-best-practices/sb-append-char](https://docs.datadoghq.com/code_analysis/static_analysis_rules/java-best-practices/sb-append-char/) static analysis rule.

# Additional Notes

# Contributor Checklist

- [ ] Format the title [according the contribution guidelines](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#labels)
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.  
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
- [ ] Update the [public documentation](https://docs.datadoghq.com/tracing/trace_collection/library_config/java/) in case of new configuration flag or behavior

Jira ticket: [PROJ-IDENT]

### CODE REVIEW

**Review by PerfectSlayer — APPROVED** [review]
I hope you used a tool to make all the changes 😅

**Comment by smola** [comment]
@PerfectSlayer Sure, just used `sed` :wink: 

Anyway, the debugger benchmarks will need to be checked, these results are weird.

File: dd-java-agent/instrumentation/testng/src/main/java/datadog/trace/instrumentation/testng/TestNGUtils.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ public static String getParameters(Object[] parameters) {
           .append(Strings.escapeToJson(String.valueOf(parameters[i])))
           .append("\"");
```
Comment (PerfectSlayer) [inline dd-java-agent/instrumentation/testng/src/main/java/datadog/trace/instrumentation/testng/TestNGUtils.java:84]: So using `sed` is why you did not catch this one for example :)

Comment (smola) [reply dd-java-agent/instrumentation/testng/src/main/java/datadog/trace/instrumentation/testng/TestNGUtils.java]: Good catch.

**Comment by smola** [comment]
Ack. I'll merge this as a one-off, since they won't be harmful anyway.

**Pull request merged**